modal/network_file_system.py

```diff
@@ -343,12 +342,16 @@ def gen_transfers():
                 relpath_str = subpath.relative_to(_local_path).as_posix()
                 yield subpath, PurePosixPath(remote_path, relpath_str)
 
-        transfer_paths = aiostream.stream.iterate(gen_transfers())
-        await aiostream.stream.map(
-            transfer_paths,
-            aiostream.async_(lambda paths: self.add_local_file(paths[0], paths[1], progress_cb)),
-            task_limit=20,
-        )
+        async with aclosing(sync_or_async_iter(gen_transfers())) as transfer_paths:
+            async with aclosing(
+                async_map(
+                    transfer_paths,
+                    lambda paths: self.add_local_file(paths[0], paths[1], progress_cb),
```

Comment:
An alternative to supporting blocking functions in the async_map (which I commented on above) would be to make sure we always pass it async functions instead, like the previous implementation did here. You could replace the lambda with an async func that just returns `self.add_local_file(paths[0], paths[1], progress_cb)` (this is the same as the `aiostream.async_`

Reply:
I see now that `add_local_file` is actually a coroutine function, so this actually works now as is since you changed the `iscoroutinefunction` condition 👍